src/main/java/no/fint/portal/admin/service/ApiDiscoveryService.java

```diff
@@ -52,7 +53,8 @@ public ApiDiscoveryService(
                     .map(UriComponentsBuilder::fromHttpUrl)
                     .map(UriComponentsBuilder::build)
                     .map(UriComponents::getPath)
-                    .map(s -> StringUtils.removeStart(s, componentPath))
+                    .map(c -> new ComponentClass(StringUtils.substringAfterLast(c, "/"), c))
```

Comment:
This might fail if the component path contains more slashes than expected - removing `componentPath` from the start is a safer option in my opinion.